File: TwoPointers/1-TwoSum.java
// problem link: https://leetcode.com/problems/two-sum/

package TwoPointers;

import java.util.HashMap;

class solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer, Integer> seen = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; i++) {

            if (seen.containsKey(target - nums[i])) {
                return new int[] { i, seen.get(target - nums[i]) };
            }

            seen.put(nums[i], i);

        }

        return new int[] { 0, 0 };

    }
}

File: ArraysHashing/1-TwoSum.cpp
// problem link: https://leetcode.com/problems/two-sum/solutions/127810/two-sum/

#include <unordered_map>
#include <vector>

class Solution {  // 2m beats 76%
 public:
  std::vector<int> twoSum(std::vector<int>& nums, int target) {
    std::unordered_map<int, int> map;

    for (int i = 0; i < nums.size(); i++) {
      if (map.count(target - nums[i]) > 0) {
        return {map[target - nums[i]], i};
      }
      map[nums[i]] = i;
    }

    return {-1, -1};
  }
};